cdap-watchdog/src/main/java/io/cdap/cdap/logging/clean/LogCleaner.java

```diff
@@ -70,5 +90,95 @@ public void run() {
     long completionTime = System.currentTimeMillis();
     LOG.info("File cleanup completed, Successful file deletes - {}. Failed file deletes - {}. Log Cleanup took {} ms",
              deleteCount, failureCount, (completionTime - startTime));
+
+    return folderCleanupCount < folderCleanupBatchSize ? DEFAULT_DELAY_IN_MILLIS : FOLDER_CLEANUP_DELAY_IN_MILLIS;
+  }
+
+  private void cleanupLogFolders() {
+    LOG.info("Starting log folder cleanup");
+    try {
+      long startTime = System.currentTimeMillis();
+      LOG.debug("Starting log folder cleanup");
+      folderCleanupCount = 0;
+
+      // TODO: (CDAP-19437): Exclude recent log folders during cleanups in LogCleaner service.
+      // For now exclude folders for current date and current date - 1.
+      long currentTime = System.currentTimeMillis();
+      String currentDate = new SimpleDateFormat(LogFileManager.FOLDER_DATE_FORMAT).format(new Date(currentTime));
+      long prevTime = System.currentTimeMillis() - MILLISECONDS_IN_DAY;
+      String prevDate = new SimpleDateFormat(LogFileManager.FOLDER_DATE_FORMAT).format(new Date(prevTime));
+      Set<String> excludeFolders = new HashSet<String>() {{
+        add(currentDate);
+        add(prevDate);
+      }};
+
+      // Clean up empty folders
+      cleanupEmptyFolders(logsDirectoryLocation, excludeFolders);
+      LOG.debug("Completed cleaning up {} log folders in {} ms", folderCleanupCount,
+          System.currentTimeMillis() - startTime);
+    } catch (IOException ex) {
+      LOG.warn("Exception while deleting log folders", ex);
+    }
+  }
+
+  /**
+   * Traverses folders in Depth-First manner and recursively deletes the folder in bottom-up manner
+   * if it is empty or sub folders are empty.
+   *
+   * @param location Root folder to start the traversal.
+   * @return Returns false if the folder cannot be cleaned up if it has files or maximum number of
+   *         folders are cleaned up, true otherwise.
+   * @throws IOException
+   */
+  private boolean cleanupEmptyFolders(@Nullable Location location, Set<String> excludeFolders) throws IOException {
+    if (excludeFolders.contains(location.getName())) {
+      // No need to traverse excluded folders
+      return false;
+    }
+    if (folderCleanupCount >= folderCleanupBatchSize) {
+      return false;
+    }
+    if (location == null) {
+      return true;
+    }
+    if (!location.isDirectory()) {
+      // Directory has files.
```

Comment:
The comment seems not matching with the logic.

Reply:
Fixed